File: app/src/main/java/com/ebsoft/watchlist/ui/create/CreateWatchlistModule.java
package com.ebsoft.watchlist.ui.create;

import com.ebsoft.watchlist.data.DataManager;

import dagger.Module;
import dagger.Provides;

/**
 * Created by endre on 21/09/18.
 */

@Module
public class CreateWatchlistModule {

    @Provides
    CreateWatchlistViewModel provideCreateWatchListViewModel(DataManager DataManager) {
        return new CreateWatchlistViewModel(DataManager);
    }
}

File: app/src/main/java/com/ebsoft/watchlist/ui/create/CreateWatchlistModule.kt
package com.ebsoft.watchlist.ui.create

import com.ebsoft.watchlist.data.DataManager
import com.ebsoft.watchlist.databinding.FragmentCreateWatchlistBinding

import dagger.Module
import dagger.Provides
import dagger.Reusable

/**
 * Created by endre on 21/09/18.
 */

@Module
class CreateWatchlistModule {
    @Provides
    @Reusable
    internal fun providesBinding(view: CreateWatchlistFragment): FragmentCreateWatchlistBinding {
        return FragmentCreateWatchlistBinding.inflate(view.layoutInflater)
    }

    @Provides
    @Reusable
    internal fun provideCreateWatchListViewModel(DataManager: DataManager): CreateWatchlistViewModel {
        return CreateWatchlistViewModel(DataManager)
    }
}
